aiodiscover/discovery.py

```diff
@@ -140,10 +140,14 @@ async def _async_get_nameservers(
         all_nameservers = list(sys_network_data.nameservers)
         router_ip = sys_network_data.router_ip
         assert router_ip is not None
-        if router_ip not in all_nameservers:
+        router_subnet = ipaddress.ip_network(f"{router_ip}/24", strict=False)
```

Comment:
There is no guarantee here that the subnet size is a /24

This would need to handle all potential subnet sizes